cli/azd/cmd/pipeline.go

```diff
@@ -68,6 +68,7 @@ func (p *pipelineConfigAction) SetupFlags(
 	local.StringVar(&p.manager.PipelineServicePrincipalName, "principal-name", "", "The name of the service principal to use to grant access to Azure resources as part of the pipeline.")
 	local.StringVar(&p.manager.PipelineRemoteName, "remote-name", "origin", "The name of the git remote to configure the pipeline to run on.")
 	local.StringVar(&p.manager.PipelineRoleName, "principal-role", "Contributor", "The role to assign to the service principal.")
+	local.StringVar(&p.manager.PipelineProvider, "provider", "", "The pipeline provider to use (GitHub and Azdo supported).")
```

Comment:
This helptext should be clearer on allowed values. `The pipeline provider to use (allowed values are: github, azdo)`. Does it also make more sense to do the free-form user input-string validation here? And convert to an enum value before the business logic `DetectProviders` run? 

We could use library support for [enumflags](https://pkg.go.dev/github.com/thediveo/enumflag).

Reply:
The `--provider` arg is the last overriding mode to set the pipeline provider.

I would not invest too much time for this at this point. Better to wait for customer's feedback here.

PM is expecting customers to use `azure.yaml` to set the pipeline provider (we have squema there).
The second most common way to set the provider is by just keeping either .azdo or .github folder (no argument required in that case)

So, using `--provider` arg is something that we might end up removing in the future.  

Note also that folks can do `--provider GitHub`  or `--provider AZDO` and `azd` would get it.. because we do lower case.